operations/HandlerChains.ts

```diff
@@ -0,0 +1,254 @@
+import {
+    CommandPlan,
+    CommandRespondable,
+    HandleCommand,
+    HandlerContext,
+    HandleResponse,
+    Instruction,
+    Response,
+} from "@atomist/rug/operations/Handlers";
+
+export type ChainedResponseHandler =
+    (response?: Response<any>, params?: {}) => ChainedInstruction;
+
+export interface HandlerChainDescriptor {
+    handle: ChainedResponseHandler;
+    onAny?: HandlerChainDescriptor;
+    onSuccess?: HandlerChainDescriptor;
+    onError?: HandlerChainDescriptor;
+}
+
+export class HandlerChain {
+    private any: HandlerChain;
+    private success: HandlerChain;
+    private error: HandlerChain;
+    constructor(private handle: ChainedResponseHandler) { }
+
+    public onAny(builder: HandlerChain): this {
+        this.any = builder;
+        return this;
+    }
+    public onSuccess(builder: HandlerChain): this {
+        this.success = builder;
+        return this;
+    }
+    public onError(builder: HandlerChain): this {
+        this.error = builder;
+        return this;
+    }
+    public build(): HandlerChainDescriptor {
+        const chain: HandlerChainDescriptor = { handle: this.handle };
+        if (this.any !== undefined) {
```

Comment:
no. People can always explicitly set things to null, or other values that don't work, and we don't guard against this. They will get errors if they do however. Do you think we need to?